src/explore-education-statistics-common/src/components/form/__tests__/FormComboBox.test.tsx

```diff
@@ -1,804 +1,812 @@
-      const { container, getByLabelText } = render(
+    test('pressing ArrowUp focuses the list box', async () => {
+      render(
         <FormComboBox
           id="test-combobox"
           inputLabel="Choose option"
-          onInputChange={() => {}}
-          onSelect={() => {}}
+          onInputChange={noop}
+          onSelect={noop}
           options={['Option 1', 'Option 2', 'Option 3']}
         />,
       );
 
-      const input = getByLabelText('Choose option') as HTMLInputElement;
+      await userEvent.type(screen.getByLabelText('Choose option'), 'Test');
 
-      fireEvent.change(input, {
-        target: {
-          value: 'Test',
-        },
-      });
-
-      const listBox = container.querySelector('[role="listbox"]');
+      const listBox = screen.getByRole('listbox');
 
       expect(listBox).not.toHaveFocus();
 
-      fireEvent.keyDown(getByLabelText('Choose option'), { key: 'ArrowUp' });
+      fireEvent.keyDown(screen.getByLabelText('Choose option'), {
+        key: 'ArrowUp',
+      });
 
-      expect(listBox).toHaveFocus();
+      const option = screen.getByText('Option 3');
+      expect(option).toHaveFocus();
     });
 
-    test('pressing ArrowUp selects the last list item', () => {
-      const { getByLabelText, getByText } = render(
+    test('pressing ArrowUp selects the last list item', async () => {
+      render(
         <FormComboBox
           id="test-combobox"
           inputLabel="Choose option"
-          onInputChange={() => {}}
-          onSelect={() => {}}
+          onInputChange={noop}
+          onSelect={noop}
           options={['Option 1', 'Option 2', 'Option 3']}
         />,
       );
+      await userEvent.type(screen.getByLabelText('Choose option'), 'Test');
 
-      const input = getByLabelText('Choose option') as HTMLInputElement;
-
-      fireEvent.change(input, {
-        target: {
-          value: 'Test',
-        },
-      });
-
-      const option1 = getByText('Option 1');
-      const option2 = getByText('Option 2');
-      const option3 = getByText('Option 3');
+      const option1 = screen.getByText('Option 1');
+      const option2 = screen.getByText('Option 2');
+      const option3 = screen.getByText('Option 3');
 
       expect(option1).toHaveAttribute('aria-selected', 'false');
+      expect(option1).not.toHaveFocus();
+
       expect(option2).toHaveAttribute('aria-selected', 'false');
+      expect(option2).not.toHaveFocus();
+
       expect(option3).toHaveAttribute('aria-selected', 'false');
+      expect(option3).not.toHaveFocus();
 
-      fireEvent.keyDown(getByLabelText('Choose option'), { key: 'ArrowUp' });
+      fireEvent.keyDown(screen.getByLabelText('Choose option'), {
+        key: 'ArrowUp',
+      });
 
-      expect(option1).toHaveAttribute('aria-selected', 'false');
-      expect(option2).toHaveAttribute('aria-selected', 'false');
-      expect(option3).toHaveAttribute('aria-selected', 'true');
+      await waitFor(() => {
```

Comment:
Do we need this `waitFor` anymore? As we discussed, it seems like the ref-based focus method seems to work better so don't think we need it?

Would also do it in any other place where we've done something similar to tidy up the test code a bit.